LibVLCSharp/Shared/MediaSource.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace LibVLCSharp.Shared
+{
+    /// <summary>
+    /// Source for VideoView that encapsulates the <see cref="LibVLC"/> and <see cref="MediaPlayer"/> instances />
+    /// </summary>
+    public class MediaSource : ISource
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="MediaSource"/> class
+        /// </summary>
+        /// <param name="cliOptions">command line options (https://wiki.videolan.org/VLC_command-line_help/)</param>
+        protected MediaSource(params string[] cliOptions)
+        {
+            if (CoreInitialized.Value)
+            {
+                LibVLC = new LibVLC(cliOptions);
+                MediaPlayer = new MediaPlayer(LibVLC);
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="MediaSource"/> class
+        /// </summary>
+        /// <param name="uri">URI of the media to play</param>
+        /// <param name="cliOptions">command line options (https://wiki.videolan.org/VLC_command-line_help/)</param>
+        protected MediaSource(string uri, params string[] cliOptions) : this(cliOptions)
+        {
+            MediaPlayer.Media = new Media(LibVLC, uri, Media.FromType.FromLocation);
+        }
+
+        /// <summary>
+        /// Finalizer
+        /// </summary>
+        ~MediaSource()
+        {
+            Dispose();
+        }
+
+        private static Lazy<bool> CoreInitialized { get; } = new Lazy<bool>(() =>
+        {
+            Core.Initialize();
+            return true;
+        });
+
+        /// <summary>
+        /// Gets the <see cref="MediaPlayer"/> object
+        /// </summary>
+        public MediaPlayer MediaPlayer { get; }
+
+        /// <summary>
+        /// Gets the <see cref="LibVLC"/> object
+        /// </summary>
+        public LibVLC LibVLC { get; }
+
+#if WINDOWS
+        public IntPtr Hwnd
+        {
+            set { MediaPlayer.Hwnd = value; }
+        }
+#elif ANDROID
+        public void SetAndroidContext(IntPtr aWindow)
+        {
+            MediaPlayer.SetAndroidContext(aWindow);
+        }
+#elif COCOA
+        public IntPtr NsObject
+        {
+            set { MediaPlayer.NsObject = value; }
+        }
+#endif       
```

Comment:
@jeremyVignelles mentioned that we don't really need those helpers as the properties are public and already `#if def'ed`. I agree with him.

Reply:
I need these properties because it's the implementation of the ISource interface. But, I'm going to remove the `public` keyword (replacing by the explicit interface implementation `IntPtr ISource.Hwnd`, ...).